chaco/tools/range_selection.py

```diff
@@ -676,13 +677,7 @@ def _determine_axis(self):
             else:
                 return 0
 
-    def __mapper_changed(self, event):
-        self.deselect()
-
-    def _axis_changed(self, old, new):
-        if old is not None:
-            self.plot.observe(
```

Comment:
So `plot` is `_plot` if it exists or else it is `component`. If we're hooking things up directly to `component`, then we're going to have bugs if `plot` property i.e. `_plot` is ever set.

Reply:
i guess we can use `trait` from observe and hook up `_plot.index_mapper` and `_plot.value_mapper` if `_plot` exists.

Actually, what happens if we listen to `plot.index_mapper` and `plot.value_mapper`?

Reply:
> Actually, what happens if we listen to plot.index_mapper and plot.value_mapper?

In this case the added test `test_notifiers_connected` fails with `deselect` never being called

Reply:
I added `_plot.index_mapper` and `_plot.value_mapper` to the observe decorator, along with a test (which previously failed) for checking that if we do explicitly set `plot`, we still get notified of changes.